bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/AbstractConfiguration.java

```diff
@@ -85,7 +85,6 @@
     protected static final String STORE_SYSTEMTIME_AS_LEDGER_UNDERREPLICATED_MARK_TIME =
             "storeSystemTimeAsLedgerUnderreplicatedMarkTime";
     protected static final String STORE_SYSTEMTIME_AS_LEDGER_CREATION_TIME = "storeSystemTimeAsLedgerCreationTime";
-    protected static final String MAX_LEDGER_METADATA_FORMAT_VERSION = "maxLedgerMetadataFormatVersion";
```

Comment:
I miss one piece of this new story.
How does the use configure this field? In order to force an older version because of the presence of a mix of old and new clients in the cluster?

Reply:
But if I start a cluster with some v3 client, and then I need to add v2 clients (which cannot upgrade bk client library version) I will fail.
We should have a way to start a cluster and make it use v2 for some time and eventually update to v3.

I have clusters shared by different applications and they cannot be upgraded all together for instance

Reply:
So, this is something that we need to decide when we push the PR which bumps the version. Should new clusters default to version 3 or default to version 2. I'm inclined to think they should default to version 2 for a couple of versions anyhow, but have version 3 available.

Reply:
I can leave with default = v2 and optional version 3 by now.
We should have a story like the journal format version and storage version